File: Goland_Python3_Algorithm/LeetCode_Daily/2024_07/12_2974_easy.py
from bisect import bisect_left, bisect_right, insort_left, insort_right, insort, bisect
from math import ceil, floor, pow, gcd, sqrt, log10, fabs, fmod, factorial, inf, pi, e
from heapq import heapify, heapreplace, heappush, heappop, heappushpop, nlargest, nsmallest
from collections import defaultdict, Counter, deque
from itertools import permutations, combinations, combinations_with_replacement, accumulate, count, groupby, pairwise
from queue import PriorityQueue, Queue, LifoQueue
from functools import lru_cache, cache, reduce
from typing import List, Optional
import sys

from sortedcontainers import SortedList, SortedDict, SortedSet

sys.setrecursionlimit(10001000)

MOD = int(1e9 + 7)
INF = int(1e20)
INFMIN = float('-inf')  # 负无穷
INFMAX = float('inf')  # 正无穷
PI = 3.141592653589793
direc = [(1, 0), (0, 1), (-1, 0), (0, -1)]
direc8 = [(1, 0), (0, 1), (-1, 0), (0, -1), (1, 1), (1, -1), (-1, 1), (-1, -1)]
ALPS = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
alps = 'abcdefghijklmnopqrstuvwxyz'

# @Author  : https://github.com/hakusai22
# @Time    : 2024/07/13 01:43
# @题目     : https://leetcode.cn/problems/minimum-number-game/description/?envType=daily-question&envId=2024-07-12
# @参考     :   https://leetcode.cn/problems/minimum-number-game/?envType=daily-question&envId=2024-07-12
# 时间复杂度 : 时间复杂度：O(nlogn)，其中 n 为 nums 的长度。瓶颈在排序上。

class Solution:
    def numberGame(self, nums: List[int]) -> List[int]:
        nums.sort()
        for i in range(1, len(nums), 2):
            nums[i - 1], nums[i] = nums[i], nums[i - 1]
        return nums

File: Goland_Python3_Algorithm/LeetCode_Daily/2024_07/12_2974_easy.go
package _024_07

import "slices"

/*
   @Author  : https://github.com/hakusai22
   @Time    : 2024/07/13 01:43
   @题目     : https://leetcode.cn/problems/minimum-number-game/solutions/2577915/pai-xu-hou-liang-liang-jiao-huan-pythonj-udxc/?envType=daily-question&envId=2024-07-12
   @参考     : https://leetcode.cn/problems/minimum-number-game/solutions/2577915/pai-xu-hou-liang-liang-jiao-huan-pythonj-udxc/?envType=daily-question&envId=2024-07-12
   @时间复杂度:
*/

func numberGame(nums []int) []int {
	slices.Sort(nums)
	for i := 1; i < len(nums); i += 2 {
		nums[i-1], nums[i] = nums[i], nums[i-1]
	}
	return nums
}
